app/mender.go

```diff
@@ -682,19 +685,39 @@ func (m *Mender) InventoryRefresh() error {
 		{Name: "mender_client_version", Value: conf.VersionString()},
 	}
 
-	if idata == nil {
-		idata = make(client.InventoryData, 0, len(reqAttr))
+	if iData == nil {
+		iData = make(client.InventoryData, 0, len(reqAttr))
 	}
-	_ = idata.ReplaceAttributes(reqAttr)
+	_ = iData.ReplaceAttributes(reqAttr)
 
-	if idata == nil {
-		log.Infof("No inventory data to submit")
+	if iData == nil {
+		log.Info("No inventory data to submit")
 		return nil
 	}
 
-	err = ic.Submit(m.api.Request(m.authToken, nextServerIterator(m), reauthorize(m)), m.Config.Servers[0].ServerURL, idata)
+	// sort slice to have a reproducible hash
+	sort.Sort(client.ByName(iData))
+
+	iDataBuf, err := json.Marshal(iData)
+	if err != nil {
+		return err
+	}
+	iDataSha := sha256.Sum256(iDataBuf)
+	lastInvSha, err := m.GetCurrentInventoryHash()
+	if err == nil && iDataSha == lastInvSha {
+		log.Info("Inventory data has not changed, skip submit")
+		return nil
+	}
+
+	log.Debugf("New inventory data, updating database with hash: %v , inventory: %v", iDataSha, iData)
```

Comment:
Question:
Cant the hash be formatted with:
`%x` ? And the inventory be stringified?

Reply:
... or even `%064x`?

Reply:
Hehe, `%v` is the lazy way...

I'll rework it a bit (at the end of the day, the main goal was to practice golang!)

Reply:
Yes, `%v` is generally discouraged because it silences all warnings about type mismatch, even useful ones. We have plenty of examples where we have made mistakes here.

Reply:
Fixed now.

Reply:
Should it not be `%32x` ? Or am I being schtoopid here?